compiler/bindings/c/iree/compiler/embedding_api.h

```diff
@@ -248,6 +248,7 @@ ireeCompilerInvocationSetVerifyIR(iree_compiler_invocation_t *inv, bool enable);
 enum iree_compiler_pipeline_t {
   IREE_COMPILER_PIPELINE_STD = 0,
   IREE_COMPILER_PIPELINE_HAL_EXECUTABLE = 1,
+  IREE_COMPILER_PIPELINE_PRECOMPILE = 2,
```

Comment:
Bike shedding this a bit...

This is breaking the main compilation into two: pre-fusion global optimizations and fusion/backend. Both of them together would comprise the "std" mode above. I'm not thrilled with the names but do agree this is a useful cut point.

We may really want a content based way of invoking like -X in clang. Possibly even selected based on a content attribute on the module (would help reproducers and such).

I'm ok with this as it is proposed for now so long as we are willing to change it later. Can you please mark this as experimental and pending a more cohesive approach for managing compilation phases in a comment.

Reply:
Added a comment.